src/models/user.model.ts

```diff
@@ -0,0 +1,147 @@
+import { DynamoDbSchema, DynamoDbTable, QueryOptions, ScanOptions } from '@aws/dynamodb-data-mapper'
+import mapper from '../db/dynamoDbClient'
+import { v4 as uuid } from 'uuid'
+import { Number } from 'aws-sdk/clients/iot'
+import MapBoxService from '../services/mapbox'
+
+const USER_NOT_FOUND = 'Not able to find user: '
+const USER_NOT_SAVED = 'Not able to save user: '
+const USER_NOT_DELETED = 'Not able to delete user: '
+
+class User {
+  id: String | undefined
+  name: String | undefined
+  dob: Date | undefined
+  address: String | undefined
+  description: String | undefined
+  createdAt: Date | undefined
+  updateAt: Date | undefined
+  imageUrl: String | undefined
+
+  constructor(name?: String,
+    dob?: Date,
+    address?: String,
+    description?: String,
+    imageUrl?: String) {
+    const currentDate: Date = new Date()
+    this.id = uuid()
+    this.name = name
+    this.dob = dob
+    this.address = address
+    this.description = description
+    this.createdAt = currentDate
+    this.updateAt = currentDate
+    this.imageUrl = imageUrl
+  }
+  save(): Promise<this> {
+    return mapper.put(this).catch((err) => Promise.reject(USER_NOT_SAVED + err))
+  }
+
+  static update(id: String, name?: String,
+    dob?: Date,
+    address?: String,
+    description?: String,
+    imageUrl?: String): Promise<User> {
+
+    return this.find(id).then((user) => {
+      user.name = name !== undefined ? name : user.name
```

Comment:
do you see a better way of doing this with the power of Typescript?

Reply:
Yes, using the Nullish coalescing operator (??)